engine/src/main/java/org/terasology/rendering/opengl/FrameBuffersManager.java

```diff
@@ -41,20 +41,35 @@
 import java.util.Date;
 import java.util.Map;
 
-import static org.lwjgl.opengl.EXTFramebufferObject.glDeleteFramebuffersEXT;
-import static org.lwjgl.opengl.EXTFramebufferObject.glDeleteRenderbuffersEXT;
-
 /**
- * The Default Rendering Process class.
+ * The FrameBuffersManager generates and maintains a number of Frame Buffer Objects (FBOs) used throughout the
+ * rendering engine.
  *
+ * In most instances Frame Buffers can be thought of as 2D arrays of pixels in GPU memory: shaders write to them or
+ * read from them. Some buffers are static and never change for the lifetime of the manager. Some buffers are dynamic:
+ * they get disposed and regenerated, i.e. in case the display resolution changes. Some buffers hold intermediate
+ * steps of the rendering process and the content of one buffer, "sceneFinal", is eventually sent to the display.
+ * <br/>
+ * At this stage no buffer can be added or deleted: the list of buffers and their characteristics is hardcoded.
+ * <br/>
+ * The existing set of public methods is primarily intended to allow communication between this manager and other parts
+ * of the rendering engine, most notably the PostProcessor and the GraphicState instances and the shaders system.
+ * <br/>
+ * An important exception is the takeScreenshot() method which prompts the renderer to eventually (not immediately)
+ * redirect its output to a file. This is the only public method that is intended to be used from outside the
+ * rendering engine.
  */
-public class LwjglRenderingProcess {
+// TODO: should most of the public methods really be as much as possible package-private?
```

Comment:
This is a question for debate. Some of the public methods are used by other (nearby) packages, but a number of methods are used only within the same package and perhaps it would be good to close them to external use?